app/src/main/java/org/oppia/android/app/spotlight/SpotlightFragment.kt

```diff
@@ -0,0 +1,413 @@
+package org.oppia.android.app.spotlight
+
+import android.content.Context
+import android.content.res.Resources
+import android.util.DisplayMetrics
+import android.util.Log
+import android.view.View
+import android.view.ViewGroup
+import android.view.animation.DecelerateInterpolator
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import com.takusemba.spotlight.OnSpotlightListener
+import com.takusemba.spotlight.OnTargetListener
+import com.takusemba.spotlight.Spotlight
+import com.takusemba.spotlight.Target
+import com.takusemba.spotlight.shape.Circle
+import com.takusemba.spotlight.shape.RoundedRectangle
+import com.takusemba.spotlight.shape.Shape
+import org.oppia.android.R
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.app.model.SpotlightViewState
+import org.oppia.android.app.onboarding.SpotlightNavigationListener
+import org.oppia.android.databinding.BottomLeftOverlayBinding
+import org.oppia.android.databinding.BottomRightOverlayBinding
+import org.oppia.android.databinding.TopLeftOverlayBinding
+import org.oppia.android.databinding.TopRightOverlayBinding
+import org.oppia.android.domain.spotlight.SpotlightStateController
+import org.oppia.android.util.accessibility.AccessibilityServiceImpl
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.data.DataProviders.Companion.toLiveData
+import java.util.Locale
+import javax.inject.Inject
+import kotlin.collections.ArrayList
+
+class SpotlightFragment @Inject constructor(
+  private val activity: AppCompatActivity,
+  private val spotlightStateController: SpotlightStateController,
+  private val accessibilityServiceImpl: AccessibilityServiceImpl
+) : Fragment(), SpotlightNavigationListener {
+  private var targetList = LinkedList<Target>()
+  private var spotlightTargetList = ArrayList<SpotlightTarget>()
+  private lateinit var spotlight: Spotlight
+  private var screenHeight: Int = 0
+  private var screenWidth: Int = 0
+  private lateinit var anchorPosition: AnchorPosition
+  private lateinit var overlayBinding: Any
+  private var internalProfileId: Int = -1
+  private var isSpotlightActive = false
+  private var isRTL = false
+
+  private fun calculateScreenSize() {
+    val displayMetrics = DisplayMetrics()
+    activity.windowManager.defaultDisplay.getMetrics(displayMetrics)
+
+    screenHeight = displayMetrics.heightPixels
+    screenWidth = displayMetrics.widthPixels
+  }
+
+  fun initialiseTargetList(spotlightTargets: ArrayList<SpotlightTarget>, profileId: Int) {
+    spotlightTargetList = spotlightTargets
+    internalProfileId = profileId
+  }
+
+  fun setInternalId(profileId: Int) {
+    internalProfileId = profileId
+  }
+
+  // since this fragment does not have any view to inflate yet, all the tasks should be done here.
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+
+    if (accessibilityServiceImpl.isScreenReaderEnabled()) {
```

Comment:
this should be moved to checkSpotlightViewState() since screen reader may be enabled anytime